### PR TITLE
Setting ErrorMessageResourceName for DataTypeAttribute by convention, #8

### PR DESCRIPTION
**Pull request by rus-vladimir**

### CODE REVIEW

**Comment by Haacked** [comment]
Thanks for the submission @rusanov-vladimir!

File: ModelMetadataExtensions/ConventionalModelMetadataProvider.cs [hunk 1]
```diff
@@ -92,10 +92,11 @@ public ConventionalModelMetadataProvider(bool requireConventionAttribute, Type d
...
             string propertyName, Type defaultResourceType)
         {
-            foreach (
-                ValidationAttribute validationAttribute in attributes.Where(a => (a as ValidationAttribute != null)))
+            foreach (var validationAttribute in attributes.OfType<ValidationAttribute>())
             {
-                if (string.IsNullOrEmpty(validationAttribute.ErrorMessage))
+                if (string.IsNullOrEmpty(validationAttribute.ErrorMessage)
+                    || validationAttribute is DataTypeAttribute
+                    && validationAttribute.ErrorMessage == ((ValidationAttribute)Activator.CreateInstance(validationAttribute.GetType())).ErrorMessage)
```
Comment (Haacked) [inline ModelMetadataExtensions/ConventionalModelMetadataProvider.cs]: It's conceivable that `CreateInstance` could throw an exception. To be safe, we should wrap that in a try/catch.

Comment (rusanov-vladimir) [reply ModelMetadataExtensions/ConventionalModelMetadataProvider.cs]: Yeap, I agree, moved extraction of the defaultErrorMessage to a separated method wrapped in try/catch.

**Comment by Haacked** [comment]
Thanks a lot! I just need to find time to publish these changes. :smile:

**Comment by haacked** [comment]
I just pushed this to Nuget. It's being indexed. Someday I'll automate this.

**Pull request merged**